### PR TITLE
Add rule to detect duplicated entry in enum

### LINKED ISSUE #1478: Report duplicated values in enum
**User story.**
As an API Designer, when I define an enumeration with duplicated value, then I want that a warning is raised.

**Describe the solution you'd like**
When I use the following `dummy.yml` file
```yaml
openapi: 3.0.0

info:
  title: Dummy title
  description: Dummy description
  version: 1.0.0

paths:
  /dummy:
    get:
      description: Dummy description
      parameters:
        - in: header
          name: some_header
          description: Some header description.
          schema:
            type: string
            enum: [dummy, dummy, not_so_dummy]
      responses:
        "200":
          description: All is good
        "400":
          description: All is good
```

And I run the command line
```bash
yarn spectral lint .\dummy.yml
```
Then I want to have an issue that is raised as the entry `dummy` appears too times in the `some_header` parameter.

**Rule proposition**

I've already prepared the following rule so that it could be included (if found useful), in the next version of Spectral.

```yaml
  enum-duplicated-entry:
    type: validation
    severity: warn
    recommended: true
    formats: ["oas3"]
    message: "A duplicated entry in the enum was found. Error: {{error}}"
    given: $..enum
    then:
      function: schema
      functionOptions:
        schema:
          type: array
          uniqueItems: true
```

Should I open a Pull request to add it in your ruleset? :)

### PR DESCRIPTION
**Pull request by Amachua**
Fixes #1478 .

**Checklist**

- [x] Tests added / updated
- [x] Docs added / updated

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by Amachua** [comment]
@P0lip To make the repository properly worked on my computer, I've followed the instruction on your great documentation CONTRIBUTING but I've faced an issue that I resolved updating the file karma.conf.ts in the following manner:

```diff
-   browsers: ['ChromeHeadless'],
+   browsers: ['ChromeHeadlessNoSandbox'],
+
+   customLaunchers: {
+     ChromeHeadlessNoSandbox: {
+       base: 'ChromeHeadless',
+       flags: ['--no-sandbox']
+     }
+   },
`` 
Would you like me to also push this update?

**Review by philsturgeon — APPROVED** [review]

**Comment by philsturgeon** [comment]
Amazing work, thank you!

File: docs/reference/openapi-rules.md [hunk 1]
```diff
@@ -397,6 +397,44 @@ TheBadModel:
         - 8
 ```
 
+### duplicated-entry-in-enum
```
Comment (Amachua) [inline docs/reference/openapi-rules.md:400]: @philsturgeon While reviewing the file I've sent, I saw that I didn't sent the right sample on this file --> Updated with good samples :)

File: test-harness/scenarios/duplicated-entry-in-enum.oas.scenario [hunk 2]
```diff
@@ -0,0 +1,33 @@
+====test====
+Identify enum values that does not respect the specified type
```
Comment (philsturgeon) [inline test-harness/scenarios/duplicated-entry-in-enum.oas.scenario:2]:
```suggestion
Identify non-unique enum values
```

**Review by philsturgeon — APPROVED** [review]

**Comment by philsturgeon** [comment]
If that config change helps then please do.

**Comment by P0lip** [comment]
@Amachua what was the error you had? Are you on Linux?

File: test-harness/scenarios/duplicated-entry-in-enum.oas.scenario (hunk 2 above)
Comment (P0lip) [reply test-harness/scenarios/duplicated-entry-in-enum.oas.scenario:2]: Ah, oops. Missed it. Going to keep that in mind and will include your suggestion in some other PR.

**Comment by P0lip** [comment]
I'll merge the PR in the meantime. LMK what that error was exactly about, so that we can evaluate the options.

**Comment by m-mohr** [comment]
This rule is too simple and should be fixed, see #1571

**Pull request merged**